src/util/color.h

```diff
@@ -2,51 +2,264 @@
 #define COLOR_H
 
 #include "util/math.h"
+#include "util/memory.h"
 
 #include <QColor>
+#include <QMap>
 
 #define BRIGHTNESS_TRESHOLD 130
 
-// algorithm by http://www.nbdtech.com/Blog/archive/2008/04/27/Calculating-the-Perceived-Brightness-of-a-Color.aspx
-// NOTE(Swiftb0y): please suggest if I should you use other methods
-// (like the W3C algorithm) or if this approach is to to performance hungry
-// NOTE: the author did not take alpha transparency into account!
-inline int brightness(int red, int green, int blue) {
-    return static_cast<int>(sqrtf(
-        red * red * .241 +
-        green * green * .691 +
-        blue * blue * .068)
-    );
+// Map the predefined colors to another color representation of them.
+class ColorsRepresentation final {
+  public:
+    // Set a color representation for a given color
+    void setRepresentation(QColor color, QColor representation) {
+        colorNameMap[color.name()] = representation.name();
+    }
+
+    // Returns the representation of a color
+    QColor map(QColor color) const {
+        if (colorNameMap.contains(color.name())) {
+            return QColor(colorNameMap[color.name()]);
+        }
+        return color;
+    }
+
+  private:
+    QHash<QString, QString> colorNameMap;
 };
 
-inline int brightness(QColor color) {
-    return brightness(color.red(), color.green(), color.red());
-}
+namespace Color {
+    static const QString Red      = "#E6194B";
```

Comment:
static is redundant here.
I think the fastes solution is: 
constexpr QStringLiteal Red("#E6194B");

Alternative would be which takes halve of storage size but CPU to create the final QString()
constexpr QLatin1String Red("#E6194B");